File: simulation_prototype/Task.py
from random import randint


class Task:
    def __init__(self, new_customerID, new_taskType, new_payload=None):
        self._ID = randint(0, 100000)
        self._customerID = new_customerID
        self._taskType = new_taskType
        self._payload = new_payload


    def getID(self):
        return self._ID


    def getCustomerID(self):
        return self._customerID


    def getTaskType(self):
        return self._taskType

    
    def getPayload(self):
        return self._payload

File: Task.cpp
#include "Task.h"

#include <stdlib.h>

Task::Task(int new_customerID, TaskTypes new_taskType, TaskPayload *new_taskPayload)
{
    ID = rand() % 1000000;
    customerID = new_customerID;
    taskType = new_taskType;
    taskPayload = new_taskPayload;
}

int Task::getID()
{
    return ID;
}

int Task::getCustomerID()
{
    return customerID;
}

TaskTypes Task::getTaskType()
{
    return taskType;
}

TaskPayload* Task::getTaskPayload()
{
    return taskPayload;
}

Task::~Task()
{
    delete taskPayload;
}